Sources/OpenCombine/Publishers/Publishers.Decode.swift

```diff
@@ -0,0 +1,107 @@
+//
+//  Publishers.Decode.swift
+//
+//
+//  Created by Joseph Spadafora on 6/21/19.
+//
+
+extension Publishers {
+
+    public struct Decode<Upstream, Output, Coder> : Publisher where
+        Upstream : Publisher,
+        Output : Decodable,
+        Coder : TopLevelDecoder,
+        Upstream.Output == Coder.Input {
+
+        /// The kind of errors this publisher might publish.
+        ///
+        /// Use `Never` if this `Publisher` does not publish errors.
+        public typealias Failure = Error
+
+        public let upstream: Upstream
+
+        internal let decoder: Coder
+
+        public init(upstream: Upstream, decoder: Coder) {
+            self.upstream = upstream
+            self.decoder = decoder
+        }
+
+        /// This function is called to attach the specified `Subscriber`
+        /// to this `Publisher` by `subscribe(_:)`
+        ///
+        /// - SeeAlso: `subscribe(_:)`
+        /// - Parameters:
+        ///     - subscriber: The subscriber to attach to this `Publisher`.
+        ///                   once attached it can begin to receive values.
+        public func receive<Receiver: Subscriber>(subscriber: Receiver)
+            where Failure == Receiver.Failure, Output == Receiver.Input {
+                let decodeSubscriber = _Decode<Upstream, Receiver, Coder>(
+                    downstream: subscriber,
+                    decoder: decoder
+                )
+                upstream.receive(subscriber: decodeSubscriber)
+        }
+    }
+}
+
+// swiftlint:disable:next line_length
+private final class _Decode<Upstream: Publisher, Downstream: Subscriber, Coder: TopLevelDecoder>:
```

Comment:
@broadwaylamb There was no way to get this to fit in the 90 chars.  Does it have to be 90 or can we pick a slightly bigger number?  Otherwise I think ignoring this one rule here is the right thing to do.